### PR TITLE
Add telegram service notification test

### PR DESCRIPTION
**Pull request by azzilian**
Add NotificationServiceImplTest - test notification service

### CODE REVIEW

File: src/test/java/online/carsharing/service/impl/NotificationServiceImplTest.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+public class NotificationServiceImplTest {
+    private static final double CAR_DAILY_FEE = 100.0;
+    private static final String CAR_BRAND = "Brand";
+    private static final String CAR_MODEL = "Model";
+    private static final String EMAIL = "email";
+    private static final long USER_ID = 1L;
+    private static final long CAR_ID = 1L;
+
+    @InjectMocks
+    private NotificationServiceImpl notificationService;
```
Comment (nklimovych) [inline src/test/java/online/carsharing/service/impl/NotificationServiceImplTest.java:38]:
```suggestion
    private NotificationServiceImpl notificationService;
    
```

File: src/test/java/online/carsharing/service/impl/NotificationServiceImplTest.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        car.setType(Type.SUV);
+        car.setDailyFee(BigDecimal.valueOf(CAR_DAILY_FEE));
+
+        user = new User();
+        user.setId(USER_ID);
+        user.setEmail(EMAIL);
+
+        rental = new Rental();
+        rental.setRentalDate(LocalDate.of(2024, 5, 1));
+        rental.setReturnDate(LocalDate.of(2024, 5, 11));
```
Comment (nklimovych) [inline src/test/java/online/carsharing/service/impl/NotificationServiceImplTest.java]: Make these Dates as constants

File: src/test/java/online/carsharing/service/impl/NotificationServiceImplTest.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+    @Test
+    @DisplayName("Send notification after new car created")
+    void sendNotification_NewCarCreated_OK() {
+        NotificationServiceImpl notificationServiceMock =
+                Mockito.mock(NotificationServiceImpl.class);
+        notificationServiceMock.createCarNotification(car);
+        Mockito.verify(
+                notificationServiceMock,
+                times(1)).createCarNotification(car);
```
Comment (nklimovych) [inline src/test/java/online/carsharing/service/impl/NotificationServiceImplTest.java]: Magic number

File: src/test/java/online/carsharing/service/impl/NotificationServiceImplTest.java [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+                .findById(1L))
+                .thenReturn(Optional.of(user));
+        Mockito.when(carRepositoryMock
+                .modelCheckById(1L))
+                .thenReturn("Car Model");
+        notificationServiceMock
+                .createRentalNotification(rental);
+        Mockito.verify(
+                notificationServiceMock,
+                times(1))
```
Comment (nklimovych) [inline src/test/java/online/carsharing/service/impl/NotificationServiceImplTest.java]: same here

**Review by nklimovych — APPROVED** [review]

**Review by Recdt — APPROVED** [review]
GJ!

**Pull request merged**